fix(docs): remove --branch flag from git clone command for bleeding edge setup in Installation.md

louisgthier opened the pull request:
The `--branch` flag was causing a fatal error during the cloning process:  
"fatal: You must specify a repository to clone." 

### Proposed change(s)

This PR removes the `--branch` flag from the `git clone` command in `Installation.md`, which was incorrectly used and caused cloning to fail for the bleeding edge version.

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)

(Provide any relevant links here, if available)

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [x] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the changelog (if applicable)
- [x] Updated the documentation (if applicable)
- [ ] Updated the migration guide (if applicable)

### Other comments

No additional comments or changes required beyond fixing the command error in the documentation.

louisgthier committed:
fix(docs): remove --branch flag from git clone command for bleeding edge setup in Installation.md

The `--branch` flag was causing a fatal error during the cloning process:
"fatal: You must specify a repository to clone."

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Unity-Technologies/ml-agents?pullRequest=6164) <br/>All committers have signed the CLA.

louisgthier committed:
docs: correct config setting name from `conditioning_type` to `goal_conditioning_type`

Updated the documentation to reflect the correct config setting name `goal_conditioning_type` instead of `conditioning_type` in the config YAML. This ensures accuracy and prevents potential confusion for users following the documentation.

louisgthier commented:
I’ve also updated the documentation to correct the config setting name from `conditioning_type` to `goal_conditioning_type`.

miguelalonsojr reviewed (approved):
LGTM.
